### ISSUE TITLE
bug: test collection fails in 0.28.0

### ISSUE THREAD

**Issue by enitrat** [foundry-rs/starknet-foundry#2386]
### Which component is your bug related to?

snforge

### Foundry Version

0.28.0

### What operating system are you using?

MacOS

### What system architecture are you using?

arm

### What happened

I just upgraded to 0.28.0, and now my tests are not collected anymore in the utils package of https://github.com/kkrt-labs/kakarot-ssj/pulls

### Trace
```
❯ snforge test
   Compiling test(utils_unittest) utils v0.1.0 (/Users/msaug/kkrt-labs/kakarot-ssj/crates/utils/Scarb.toml)
    Finished release target(s) in 10 seconds
   Compiling utils v0.1.0 (/Users/msaug/kkrt-labs/kakarot-ssj/crates/utils/Scarb.toml)
    Finished release target(s) in 3 seconds



Collected 0 test(s) from utils package
Running 0 test(s) from src/
Tests: 0 passed, 0 failed, 0 skipped, 0 ignored, 0 filtered out
❯
❯ snforge --version
snforge 0.28.0
❯ asdf local starknet-foundry 0.27.0
❯ snforge test
   Compiling utils v0.1.0 (/Users/msaug/kkrt-labs/kakarot-ssj/crates/utils/Scarb.toml)
    Finished release target(s) in 3 seconds


Collected 261 test(s) from utils package
Running 261 test(s) from src/
```
### Is there an existing issue for this?

- [ ] I have searched the existing issues and verified no issue exits for this problem.

**Comment by enitrat** [comment]
update: I was missing the snforge_std dependency in my package.

But now, everything is even more broken:

on 0.27.0 using snforge_std 0.28.0:
scarb test -p utils works (package that uses snfoundry and `test = snforge test`)
scarb test -p contracts doesnt work (uses cairo test)

on 0.280:
nothing works

errors:


```
error: Plugin diagnostic: Attribute should only appear on tests.
 --> /Users/msaug/kkrt-labs/kakarot-ssj/crates/contracts/src/storage.cairo:183:1

^

error: Plugin diagnostic: Attribute should only appear on tests.
 --> /Users/msaug/kkrt-labs/kakarot-ssj/crates/contracts/src/storage.cairo:190:1

^

error: Plugin diagnostic: Attribute should only appear on tests.
 --> /Users/msaug/kkrt-labs/kakarot-ssj/crates/contracts/src/storage.cairo:199:1

```

Context: i'm using a workspace that relies on both snfoundry and cairo-test.  see: https://github.com/kkrt-labs/kakarot-ssj/pull/864

**Comment by cptartur** [comment]
Turns out this is issue with Scarb https://github.com/software-mansion/scarb/issues/1535

Copying the response from telegram:

> I can confirm, that there is a bug in Scarb that: if you have two packages in the same workspace, and one of the packages depend on the other, if you try to compile the first package, it will get dev-dependencies from both of them. 
Until we release a fix, you can remove `contracts` package from the workspace and it should work properly.

**Comment by piotmag769** [comment]
Not relevant anymore, no? @cptartur @enitrat

**Comment by piotmag769** [comment]
Solved with Scarb `2.8.0`
